USCSandbox/UltraShaderConverter/USIL/Metadders/USILSamplerMetadder.cs

```diff
@@ -19,7 +19,7 @@ public bool Run(UShaderProgram shader, ShaderParams shaderParams)
                         //    p => p.SamplerIndex == operand.registerIndex
                         //);
                         TextureParameter? texParam = shaderParams.TextureParameters.FirstOrDefault(
-                            p => p.Index == operand.registerIndex
+                            p => p.SamplerIndex == operand.registerIndex
                         );
 
                         if (texParam == null)
```

Comment:
Even with SamplerIndex this still seems to fail sometimes, but in those cases it seems we have a -1 SamplerIndex. Let's add that just in case:
```suggestion
                        if (texParam == null)
                        {
                            // fallback to -1 if it exists
                            texParam = shaderParams.TextureParameters.FirstOrDefault(
                                p => p.SamplerIndex == -1
                            );

                            if (texParam == null)
```